### DISCUSSION TITLE
esp32.wake_on_ext0 consistency across different chips

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#14539] [Using MicroPython]
The last time I used esp32.wake_on_ext0 was on a regular esp32 with upython v1.17 in August last year and it was working OK with both light & deep sleep.
Today I'm on an esp32s2 with upython v1.22 and it's not working with either sleep type (the board immediately wakes up from sleep as soon as it enters instead of waiting for me to press the relevant button).
Just wondering what what level of consistency to expect from  something like esp32.wake_on_ext0 across different upython versions/chips?

**Comment by kjm1102** [comment]
If I run
```
import os,esp32,time
from machine import Pin,deepsleep
print (os.uname())
pb=Pin(0, Pin.IN); esp32.wake_on_ext0(pb,0)
deepsleep()
```
on an esp32, I get 
```
MPY: soft reboot
(sysname='esp32', nodename='esp32', release='1.21.0', version='v1.21.0 on 2023-10-05', machine='Generic ESP32 module with ESP32')
```
then, after I press the button on gpio0, I get the esp32 waking up
```
ets Jul 29 2019 12:21:46

rst:0x5 (DEEPSLEEP_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4728
load:0x40078000,len:14876
ho 0 tail 12 room 4
load:0x40080400,len:3368
entry 0x400805cc
MicroPython v1.21.0 on 2023-10-05; Generic ESP32 module with ESP32
Type "help()" for more information.
```
But if I try the same thing on an esp32s2 ,I get
```
MPY: soft reboot
(sysname='esp32', nodename='esp32', release='1.22.2', version='v1.22.2 on 2024-02-22', machine='Generic ESP32S2 module with ESP32S2')
PROBLEM IN THONNY'S BACK-END: Exception while handling 'Run' (ConnectionError: EOF).
```
because the esp32s2 hasn't waited for the button press but exited deepsleep immediately.
I'm wondering why the esp32s2 does not behave the same as the esp32?

**Reply by KrTamir** [reply]
I have the same problem on VScode (termina)
would love to get a solotion.

**Reply by kjm1102** [reply]
You could try putting hold=1 in the pin definition, https://github.com/orgs/micropython/discussions/15234#discussioncomment-9751258  Regular esp32s don't need it for gpio0 but for the s2 does for some reason.

**Comment by projectgus** [comment]
Adding a note that kjm1102 has opened an issue about this here: https://github.com/micropython/micropython/issues/15260

They've also left a comment with a workaround, so I'm going to close the question in favour of the issue.
